### PR TITLE
feat: Enable pipeline failure log

### PR DESCRIPTION
**Pull request by vli11**
Closes: issue #41

_The PR review is to check for sustainability and correctness.  Sustainability is actually more business critical as correctness is largely tested into the code over time.   Its useful to keep in mind that SW often outlives the HW it was written for and engineers move from job to job so it is critical that code developed for Intel be supportable across many years.  It is up to the submitter and reviewer to look at the code from a perspective of what if we have to debug this 3 years from now after the author is no longer available and defect databases have been lost.  Yes, that happens all the time when we are working with time scales of more than 2 years.  When reviewing your code it is important to look at it from this perspective._

Author Mandatory (to be filled by PR Author/Submitter)
------------------------------------------------------
- Developer who submits the Pull Request for merge is required to mark the checklist below as applicable for the PR changes submitted.  
- Those checklist items which are not marked are considered as not applicable for the PR change.  
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the developers for Inner Source Development Model (ISDM) compliance

### PULL DESCRIPTION
_Provide a 1-2 line brief overview of the changes submitted through the Pull Request..._

### REFERENCES
Reference URL for issue tracking (Github): **\<URL to be filled>**

- [x] Added label to the Pull Request for easier discoverability and search

### CODE MAINTAINABILITY
- [x] **_Commit Message meets guidelines as indicated in the URL\*_**
	- https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [x] **_Every commit is a single defect fix and does not mix feature addition or changes\*_**
- [ ] Added required new tests relevant to the changes
	- [ ] PR contains URL links to functional tests executed with the new tests 
- [ ] Updated Documentation as relevant to the changes
- [ ] Updated Build steps/commands changes as relevant
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide description)
- [ ] Specific instructions or information for code reviewers (If any):

Maintainer Mandatory (to be filled by PR Reviewer/Approving Maintainer)
-----------------------------------------------------------------------
- Maintainer who approve the Pull Request for merge are required to mark the checklist below as appropriate for the PR change reviewed as key proof of attestation indicating reasons for merge. 
- Those checklist items which are not marked are considered as not applicable for the PR change. 
- Items marked with an asterisk suffix are mandatory items to check.

### QUALITY CHECKS
- [ ] Architectural and Design Fit
- [ ] **_Quality of code (At least one should be checked as applicable)\*_**
	- [ ] Commit Message meets guidelines
	- [ ] PR changes adhere to industry practices and standards
	- [ ] Adopted domain specific coding standards ([SBL Coding Standards](https://github.com/tianocore-docs/Docs/raw/master/Specifications/CCS_2_1_Draft.pdf))
	- [ ] Error and exception code paths implemented correctly
	- [ ] Code reviewed for domain or language specific anti-patterns
	- [ ] Code is adequately commented
	- [ ] Code copyright is correct
	- [ ] Tracing output are minimized and logic
	- [ ] Confusing logic is explained in comments
	- [ ] Commit comment can be used to design a new test case for the changes
- [ ] **_Test coverage shows adequate coverage with required CI functional tests pass on all supported platforms\*_**
- [ ] **_Static code scan report shows zero critical issues\*_**

### CODE REVIEW IMPACT
- Summary of Defects Detected in Code Review: **\<%P1*xx,P2*xx,P3*xx,P4*xx%>** \
Note P1/P2/P3/P4 denotes severity of defects found (Showstopper/High/Medium/Low) and xx denotes number of defects found

# _Code must act as a teacher for future developers_

### CODE REVIEW

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
not just --rm but also needs to clean run containers after it's done

File: docker-run.sh [hunk 1]
```diff
@@ -5,6 +5,8 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
+# clean up exited containers
+docker rm $(docker ps -a -f status=exited -q)
```
Comment (jim-wang-intel) [inline docker-run.sh:9]: only clean up those vision check containers not everything exited

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

**Review by brian-intel — APPROVED** [review]

**Comment by jim-wang-intel** [comment]
i think we also need a separate detailed log file from gst-launch of pipeline execution... the current log file pipeline1.log doesn't help very much on what's wrong if pipeline failed

File: docker-run-opencv-ovms.sh [hunk 2]
```diff
@@ -221,6 +221,6 @@ docker run --network host $cameras $TARGET_USB_DEVICE $TARGET_GPU_DEVICE --user
 -e inputsrc="$inputsrc" $RUN_MODE $stream_density_params \
 -e CPU_ONLY="$CPU_ONLY" \
 -e GRPC_PORT="$GRPC_PORT" \
--e AUTO_SCALE_FLEX_140="$AUTO_SCALE_FLEX_140" --rm $CLIENT_TAG bash -c /scripts/$pipeline
+-e AUTO_SCALE_FLEX_140="$AUTO_SCALE_FLEX_140" $CLIENT_TAG bash -c /scripts/$pipeline
```
Comment (jim-wang-intel) [inline docker-run-opencv-ovms.sh:224]: if removing --rm please also add the clean up at the start up in this script

File: docker-run.sh [hunk 3]
```diff
@@ -5,6 +5,8 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
+# clean up exited containers
+docker rm $(docker ps -a -f name=automated-self-checkout -f status=exited -q)
```
Comment (jim-wang-intel) [inline docker-run.sh:9]: this should be in another docker run script `docker-run-dlstreamer.sh` to be more encapsulated

File: docs_src/pipelinerun.md [hunk 4]
```diff
@@ -152,7 +152,14 @@ This directory also contains r*.jsonl for each of pipeline/workload that is runn
...
 !!! Failure
-    If you do not see above Docker container(s), review the console output for errors. Sometimes dependencies fail to resolve and must be run again. Address obvious issues and try again repeating the above steps.
+    If you do not see above Docker container(s), review the console output for errors. Sometimes dependencies fail to resolve and must be run again. Address obvious issues and try again repeating the above steps. Here are couple debugging tips:
+
+    1.check the docker logs using following command
+
+    ```bash
+    docker logs <containerId>
+    ```
+    2. check gst-launch log in /results/gst-launch*.log
```
Comment (jim-wang-intel) [inline docs_src/pipelinerun.md:162]: is result absolute directory with starting /?

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

File: docker-run-opencv-ovms.sh [hunk 5]
```diff
@@ -5,6 +5,12 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
+SERVER_CONTAINER_NAME="model-server"
+CLIENT_CONTAINER_NAME_PREFIX="ovms-client"
+# clean up exited containers
+docker rm $(docker ps -a -f name=$SERVER_CONTAINER_NAME -f status=exited -q)
```
Comment (jim-wang-intel) [inline docker-run-opencv-ovms.sh:11]: if you do this, please also do `removing --rm` from server container

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

File: docker-run-opencv-ovms.sh (hunk 5 above)
Comment (vli11) [reply docker-run-opencv-ovms.sh:11]: it's done and included in last commit

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**